src/Compilers/Core/Portable/CommandLine/CommonCompiler.cs

```diff
@@ -86,11 +86,17 @@ public CommonCompiler(CommandLineParser parser, string responseFile, string[] ar
 
         internal abstract bool SuppressDefaultResponseFile(IEnumerable<string> args);
 
-        internal string GetAssemblyFileVersion()
+        /// <summary>
+        /// The type of the compiler class for version information in /help and /version.
+        /// We don't simply use this.GetType() because that would break mock subclasses.
+        /// </summary>
+        internal abstract Type Type { get; }
+
+        internal virtual string GetAssemblyFileVersion()
         {
             if (_clientDirectory != null)
             {
-                var name = $"{typeof(CommonCompiler).GetTypeInfo().Assembly.GetName().Name}.dll";
+                var name = $"{Type.GetTypeInfo().Assembly.GetName().Name}.dll";
```

Comment:
I think it is worth exploring if we can use Assembly.Location to extract the name.

---
In reply to: [81196987](https://github.com/dotnet/roslyn/pull/14181#discussion_r81196987) [](ancestors = 81196987)